binding/webgl_rendering_context.cc

```diff
@@ -986,6 +987,73 @@ napi_value WebGLRenderingContext::BufferData(napi_env env,
   return nullptr;
 }
 
+/* static */
+napi_value WebGLRenderingContext::BufferSubData(napi_env env,
+                                                napi_callback_info info) {
+  LOG_CALL("BufferSubData");
+  napi_status nstatus;
+
+  size_t argc = 3;
+  napi_value args[3];
+  napi_value js_this;
+  nstatus = napi_get_cb_info(env, info, &argc, args, &js_this, nullptr);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+  ENSURE_ARGC_RETVAL(env, argc, 3, nullptr);
+
+  WebGLRenderingContext *context = nullptr;
+  nstatus = UnwrapContext(env, js_this, &context);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+
+  ENSURE_VALUE_IS_NUMBER_RETVAL(env, args[0], nullptr);
+  GLenum target;
+  nstatus = napi_get_value_uint32(env, args[0], &target);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+
+  // start here
+  uint32_t offset;
+  ENSURE_VALUE_IS_NUMBER_RETVAL(env, args[1], nullptr);
+  nstatus = napi_get_value_uint32(env, args[1], &offset);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+
+  napi_valuetype arg_valuetype;
+  nstatus = napi_typeof(env, args[2], &arg_valuetype);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+
+  if (arg_valuetype == napi_object) {
```

Comment:
gl.bufferSubData() only takes ArrayView-instances:

https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferSubData

You can check with `napi_is_arraybufferview` and `napi_get_arraybuffer_info`. Feel free to add a macro to utils.h for `ENSURE_NAPI_IS_VALUE_ARRAY_BUFFER*`

Reply:
Added `ENSURE_VALUE_IS_ARRAY_BUFFER` and `ENSURE_VALUE_IS_ARRAY_BUFFER_RETVAL` to utils.h.